docs/config.md

```diff
@@ -0,0 +1,115 @@
+# Configuration files
+
+You can describe Thinc models and experiments using a configuration format
+based on Python's built-in `configparser` module. We add a few additional
+conventions to the built-in format, to provide support for non-string values,
+support nested objects, and to integrate with Thinc's *registry system*.
+
+## Example config
+
+
+```
+[some_section]
+key1 = "string value"
+another_key = 1.0
+# Comments, naturally
+third_key = ["values", "are parsed with", "json.loads()"]
+some_other_key =  
+    {
+        "multiline values?": true
+    }
+
+[another_section]
+more_values = "yes!"
+null_values = null
+interpolation = ${some_section:third_key}
+
+# Describe nested sections with a dot notation in the section names.
+[some_section.subsection]
```

Comment:
Nitpicking: do we want a convention where subsections are in the config in order, right below their respective supersection? Because in this example case we first have `some_section`, then `another_section` and then a subsection of `some_section`. We could rewrite the example to be `[another_section.subsection]` so that the order seems more natural ?